messageapp/src/main/java/com/bitcoinminers/messageapp/Chat.java

```diff
@@ -24,6 +24,11 @@ public class Chat implements Saveable {
      */
     private ArrayList<Message> messages = new ArrayList<>();
 
+    /**
+     * Users in the chat.
+     */
+    private ArrayList<User> users = new ArrayList<>();
```

Comment:
Having users stored in both the server and group chats makes saving difficult (as one user may be 'revived" from a save twice). Maybe save user IDs in chats.